### ISSUE TITLE
[QUALITY ASSURANCE]:  Testing Compare grading page (sprint 18)

### ISSUE THREAD

**Issue by Ravirkt** [fdnd-agency/footguard#226]
- [x] Browser Test (Controleren of alles echt werkt op verschillende browsers)
- [x] Performance test
- [x] Toegankelijkheid - WCAG - a11y - screenreader - keyboard

**Comment by Ravirkt** [comment]
# Browser Test 🌐

Test uitgevoerd op de volgende browsers:  

- Safari 🧭  
- Chrome 🌏  
- Edge 🔵  
- Opera 🎤  
- Firefox 🔥  

---

## Tests voor elke browser ✅

### Testpunten
- Dynamische data  
- Styling  
- Responsiveness 
- Errorpagina verschijnt bij onjuiste parameters  

### Resultaten  
- Styling is consistent zoals het hoort
- De pagina is responsive
- De dynamische data wordt ingeladen
- De errorpagina verschijnt bij ongeldige url
- overflow bug alleen op firefox

### Demonstratie 🪧

---

## Resultaten per browser



Safari     

<img width="1875" height="1160" alt="Image" src="https://github.com/user-attachments/assets/4a6aa31e-bd3c-4c50-8291-15d1a43b8c07" />


Chrome     

<img width="1537" height="1188" alt="Image" src="https://github.com/user-attachments/assets/929fd140-5dd4-4031-b1d2-206f1663bbcd" />


Edge         

<img width="1885" height="1197" alt="Image" src="https://github.com/user-attachments/assets/e7bdfff7-00b7-4c1f-8357-bc626ca6f506" />


Opera  

<img width="1889" height="1198" alt="Image" src="https://github.com/user-attachments/assets/bcc5c7d6-83d3-4b4d-b2bc-561356462d76" />


Firefox     

https://github.com/user-attachments/assets/d17ed6b9-5725-460f-9e3c-a390328ed7df

Voor de bug is een issue aangemaakt: https://github.com/orgs/fdnd-agency/projects/84/views/1?sliceBy%5Bvalue%5D=_noValue&pane=issue&itemId=151421995&issue=fdnd-agency%7Cfootguard%7C219

**Comment by Ravirkt** [comment]
# Performance test

## Desktop

<img width="1532" height="1239" alt="Image" src="https://github.com/user-attachments/assets/dd6e1d3b-239f-4602-a0fe-42feede146c6" />

## Mobile

<img width="1704" height="1157" alt="Image" src="https://github.com/user-attachments/assets/1ac86986-e084-4765-88a2-062fa98f4612" />

**Comment by Ravirkt** [comment]
# Toegankelijkheid ♿

Test uitgevoerd op de volgende onderdelen:  

- Screenreader 📢  
- Keyboard ⌨️  
- Lighthouse 🔍  
- A11y checklist 📋  

---

## Keyboard ⌨️

### Testpunten ✅
- De gebruiker kan door de pagina **tabben** zonder probleem  
- Duidelijke focus outline zichtbaar  
- form is te bedienen met toetsenbord
- Je scrollen door de table met toetsenbord

### Resultaten 📊
- Alle interactieve elementen zijn toegankelijk met het toetsenbord  
- Tabvolgorde loopt logisch van boven naar beneden  
- De checklist kan volledig met toetsenbord worden ingevuld
- De dropdown notes kan met toetsenbord open en gesloten worden
- Je kan met de tab toets de table selecteren en vervolgens met de up en down pijlen naar boven en beneden scrollen.

---

## Lighthouse 🔍

### Testpunten ✅
- Desktop audit  
- Mobile audit  

### Resultaten 📊

#### Desktop

<img width="1520" height="1161" alt="Image" src="https://github.com/user-attachments/assets/aa3ab126-466b-44b4-bc10-6bb06a45bcd3" />


#### Mobile

<img width="1165" height="1172" alt="Image" src="https://github.com/user-attachments/assets/4da79a69-d7d2-43f2-90cf-057eb0707353" />

---

## Screenreader 📢

### Testpunten ✅
- Kan pagina correct voorlezen  
- Alle interactieve elementen zijn toegankelijk  

### Resultaten 📊
- De screenreader noemt de elementen op
- De screenreader lees de table niet op. Het zegt alleen dat het op een table gaat. Ik heb wat onderzoek gedaan en ben gekomen op de <caption> tag. Hierin heb ik een duidelijke beschrijving van de table en de content in de kolommen verwerkt. Ik heb de caption hidden gemaakt door de visually-hidden methode van a11y.
- De namen van de kolommen werden ook niet opgenoemd. Ik heb dus de html scope="col" gebruikt. Nu kan de screenreader en de keyboard tabben door de columns. ---> COMMIT WAARIN AANPASSINGEN ZIJN AANGEBRACHT: https://github.com/fdnd-agency/footguard/commit/f8bc5e14c71adbe0a4f926d78f8d9673d7b5786b
- Demo / opname:

https://github.com/user-attachments/assets/e5cdc704-a909-4958-b314-096b2ca810e1

https://github.com/user-attachments/assets/4aafc353-1966-4aa3-885b-f9b802a77756

---

## A11y Checklist 📋

### Testpunten ✅
- Content, global code, keyboard, images, headings, lists, controls, tables, forms, media, appearance & color contrast, mobile & touch  

### Resultaten 📊
### Content
- [x] Use plain language and avoid figures of speech, idioms, and complicated metaphors.
  - gecontroleerd op grammatica, engelse taal en spelling
- [x] Make sure that `button`, `a`, and `label` element content is unique and descriptive.
  - naamgeving buttons , a, en labels zijn gecontroleerd op functie doeleinde 
- [x] Use left-aligned text for LTR languages, and right-aligned text for RTL languages.
  - uitlijning tekst is correct begint van links naar rechts

### Global code
- [x] Validate your HTML.
https://github.com/fdnd-agency/footguard/issues/226#issuecomment-3795738596- [x] Use a `lang` attribute on the `<html>` element.
  - <html lang="en"> is aanwezig
- [x] Ensure that zoom is not disabled.
  - de gebruiker kan in en- en uitzoomen in de pagina
- [x] Use landmark elements (`<nav>`, `<main>`, `<footer>`).
  - footer is niet aanwezig omdat deze niet relevant is
- [x] Ensure linear content flow.
  - de interactieve elementen zijn toegankelijk op logische flow van de pagina
- [x] Avoid `autofocus` and make session timeouts extendable.
  - geen gebruik gemaakt van autofocus
- [x] Use labels instead of relying on tooltips.
  - formitems en buttons hebben label

### Keyboard
- [x] All interactive elements (links, buttons, controls) are operable via keyboard.
  - de interactieve elementen zijn toegankelijk met de toetsenbord. je kunt van boven naar beneden tabben per element. je kunt de hele form door met de toetsen
- [x] There is a visible focus-style for interactive elements.
  - De interactieve elementen hebben alle een duidelijke outline dat zichtbaar is waarop gefocust is.

### Images
- [x] All meaningful images have appropriate alt text.
  - Er zijn geen images op de pagina alleen icons
- [x] Avoid using color as the only way to convey information.
  - er is voornamelijk met een icon, text en kleur aangegeven

### Headings
- [x] Use headings in logical order (`<h1>` → `<h2>` → `<h3>`).
  - headings zijn op logische volgorde gebruikt
- [x] Do not skip heading levels.
  - headings levels zijn niet overgeslagen alles begint van h1 naar aflopend

### Lists
- [x] Use correct list markup: `<ul>`, `<ol>`, or `<dl>` where appropriate.
  - n.v.t
- [x] Ensure the list structure makes sense for assistive technologies.
  - n.v.t

### Controls
- [x] Buttons, links and form controls have clear, descriptive labels.
  - buttons en labels hebben duidelijke benaming en label naam
- [x] Interactive elements that are not standard controls must provide proper ARIA roles or labels.
  - n.v.t

### Tables
- [x] Use `<th>` for headers and ensure clear association with `<td>` cells.
  - Dit is uitgevoerd op de table
- [x] Do not use tables purely for layout.
  - de table wordt niet voor layout gebruikt maar weergeven van data

### Forms
- [x] Ensure each form control has a `<label>` (explicit or via `aria-label`).
  - de form heeft voor de inputs alle een label
- [ ] Error messages should be clearly visible and explain how to fix the error.
  - de post van de form is nog niet werkend

### Media (Video / Audio)
- [x] Provide captions or transcripts for videos and audios.
  - n.v.t
- [x] Media controls must be accessible (keyboard, screen-reader).
  - n.v.t

### Appearance & Color contrast
- [x] Color is not the only way to convey information.
  - zowel kleur als text en icons maken informatie duidelijk
- [x] Maintain sufficient contrast between text and background (WCAG AA: 4.5:1 for normal text).
  - alle kleuren zijn getest op contrast
- [x] Ensure icons and badges also meet contrast criteria.
  - de icons zijn getest op contrast

### Mobile & Touch
- [x] Touch-targets (buttons/links) are large enough and spaced adequately.
  - De interactieve elementen zijn groot genoeg en gemakkelijk klikbaar op touchscreen devices
- [x] Users can zoom and rotate screen without losing access to content.
  - Getest en werkt

**Comment by Ravirkt** [comment]
# HTML validator 🚪

<img width="1476" height="1178" alt="Image" src="https://github.com/user-attachments/assets/82c3cb0f-9dd7-404a-a370-d31c4cee8cf5" />
